!Add position option with 6 configurable positions (replaces mode)

dpi0 opened the pull request:
### BREAKING CHANGE: Replaced "mode" option with "position"

This change introduces a new `position` option, replacing the `mode` option.

#### Position Options
| Position | Integer Value |
| --- | --- |
| full screen | 0 |
| top | 1 |
| right | 2 |
| bottom | 3 |
| left | 4 |
| center | 5 |

* **Vertical positions**: `left` and `right` - `width` option applies
* **Horizontal positions**: `top` and `bottom` - `height` option applies
* **Center position**: both `height` and `width` options apply
* **Full screen position**: neither `height` nor `width` options apply

#### Usage

```lua
require("neoterm").setup({
	clear_on_run = true, -- run clear command before user specified commands
	noinsert = false, -- disable entering insert mode when opening the neoterm window
	position = "left", -- or position = 2
	width = 0.7, -- sets the width to 70%
	height = 0.4, -- doesn't apply to "left" position
})
```

dpi0 committed:
!Add position option with 6 configurable positions

Replaced "mode" with "position". Where a position can be one of 6 either
a string or integer, full screen (0), top (1), right (2), bottom (3),
left (4) and center (5).

itmecho commented on lua/neoterm.lua:
```diff
@@ -30,21 +32,63 @@ local neoterm = {}
 -- Sets global configuration
 -- Options:
---	clear_on_run - send the clear comand before running a command with run or rerun
---	mode - set how the terminal window will be displayed
---		* vertical
---		* horizonal
---		* fullscreen
```
To save a breaking change, could we set the `position`, `width`, and `height` based on this parameter? i.e. `horizontal` translates to `position = 'right', `vertical` translates to `position = 'bottom'`, and `fullscreen` translates to `position = 'fullscreen'`

If the `position` is set, it should take priority and `mode` should be ignored.

Could add a deprecated message to the mode param.

dpi0 replied:
Added a deprecated warning interrupting the user on startup if the mode option is set.

Also 33326e418501c7e0c933981083663bc012634114 adds a simple translation from the old mode option to the newer position option

itmecho commented on lua/neoterm.lua:
```diff
@@ -59,28 +103,62 @@ function neoterm.open(opts)
+
+    if type(position) == "string" then
+      position = ({
+        ["full screen"] = 0,
```
I would prefer this stayed as `fullscreen`. Comments above and the error message would need to be updated as well

```suggestion
        ["fullscreen"] = 0,
```

dpi0 replied:
eb20ccc02c405458875c3f63a047b325476eac9e fixes the "position" type warning.

Also 59c299a3224886d141382de40791b87481494714 renames "full screen" to "fullscreen"

itmecho reviewed (changes requested):
Thanks for this, it looks great! Just a couple of things to update and then I'm happy to merge this change.

itmecho commented:
Oh, we'd also need to update the docs and the README to reflect the new settings

dpi0 committed:
Rename "full screen" position to "fullscreen"

dpi0 committed:
Add backwards compatibility for "mode" option

Handles 3 options "vertical", "horizontal" and "fullscreen" and
translates them to "right", "bottom" and "fullscreen" respectively

dpi0 committed:
Add a default height for positions top, bottom and center

dpi0 committed:
Update docs with a new configuration section

dpi0 committed:
Update README.md without proper images

dpi0 committed:
Update README.md with screenshots & minor wording changes

dpi0 committed:
Fix default height not applying in top/bottom/center

dpi0 committed:
Fix warning not interrupting user when using mode option

dpi0 committed:
Fix type conversion warning "position" var

dpi0 committed:
Update README.md with fixed screenshots' height

dpi0 commented:
> Oh, we'd also need to update the docs and the README to reflect the new settings

fde3171ad8cd445cd12dfdea5dde65f5f4bd7715 updates the neoterm.txt and tags.

1d9557fe62fafc0828c08b762a2f37ea7fe07f83 updates README.md

dpi0 commented:
**SIDENOTE:** an update from `master` to `main` would be great!

itmecho reviewed (approved):
This looks great! Thanks for doing those changes
